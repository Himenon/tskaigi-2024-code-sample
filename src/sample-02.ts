import ts from "typescript";
import * as fs from "node:fs";

import * as Printer from "./utils";

const factory = ts.factory;

export const run = () => {
  const code = Printer.generateCodeByStatements([
    factory.createVariableStatement(
      undefined,
      factory.createVariableDeclarationList(
        [
          factory.createVariableDeclaration(
            factory.createIdentifier("hello"),
            undefined,
            undefined,
            factory.createArrowFunction(
              undefined,
              undefined,
              [],
              undefined,
              factory.createToken(ts.SyntaxKind.EqualsGreaterThanToken),
              factory.createBlock([factory.createReturnStatement(factory.createStringLiteral("world"))], true),
            ),
          ),
        ],
        ts.NodeFlags.Const,
      ),
    ),
  ]);

  fs.writeFileSync("output/sample2.ts", code, "utf8");

  console.log("Generated output/sample2.ts");
};
