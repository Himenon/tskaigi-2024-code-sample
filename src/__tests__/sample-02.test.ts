import * as fs from "node:fs";
import { test, expect } from "vitest";
import * as Module from "../sample-02";

test("Sample2: Code Generate Test", () => {
  Module.run();

  const generateCode = fs.readFileSync("output/sample-02.ts", {
    encoding: "utf-8",
  });
  expect(generateCode).toMatchSnapshot();
});
