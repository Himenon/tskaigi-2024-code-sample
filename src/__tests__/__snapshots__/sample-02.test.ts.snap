// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sample2: Code Generate Test 1`] = `
"const hello = () => {
    return "world";
};
"
`;
